app/graphql/types/system.rb

```diff
@@ -64,6 +69,15 @@ def rules_failed(args = {})
       end
     end
 
+    def rules_failed_severity(args = {})
+      ::RuleResult
+        .for_policy(args[:policy_id])
+        .failed
+        .joins(:rule).where(rules: { severity: args[:severity] })
+        .joins(:host).where(hosts: { id: object.id })
+        .select('rule_results.host_id', 'rules.id', 'rules.severity')
+    end
```

Comment:
There's way too many joins here and the query seems to run separately for each system. This should be batch loaded...